Update track changes section (#133)
bump-sh/docs: src/_help/changes-management/index.md

---
title: Changes management
---

- TOC
{:toc}

> Tracking API changes is essential, both for users and developers.

Reviewer: ```suggestion
Tracking API changes is essential, both for API builders and consumers.
```

## What is a change

> Changes are differentiated into various categories:

Reviewer: This doesn't explain what is a change. We should probably redefine the concept here.

Author: I'm not touching this part until it has been properly define on our end. My first draft was something around: "a change is a modification of your API definition with an impact on its behavior"

Reviewer: Final version: A change is a modification of your API with an impact on its behavior

The change is done on the API (by changing its code). The definition just reflects this change, but when we talk about a change, it's an API one, not an API definition one, imo.

| Documentation change     | a description is modified  |

> | Structural change        | the structure of the API definition is modified  |

Reviewer: ```suggestion
| Structural change        | the structure of the API is modified  |
```

> | Breaking change    | a major change in your API definition requires an update to the code consuming the API for its users |

Reviewer: ```suggestion
| Breaking change    | a major change in your API requires an update to the code consuming it |
```

Breaking changes are the ones your API consumers should not miss and are highlighted in the changelog. Here is a non-exhaustive list of criteria that help us determine if a change is breaking or not:
- Renaming/Deleting an endpoint or operation
- Renaming/Deleting a property (body, parameters, etc...)
- Changing the "type" attribute of a property
- Removing polymorphism of a property
- Marking an existing property as required

> - Add or delete a security requirement

Reviewer: This should be an exhaustive list. Also, WDYT of moving this to the `Concepts` page? I think it would be the best place to define what's a breaking change for us. We should also add `API structure change` concept, and `Documentation change`.

Author: Will stick to Breaking changes today. Will work on the last ones later.
I haven't been able to find any exhaustive list, anywhere. So I'd try to investigate further more or even request help from the team.

Reviewer: :+1:

## How to manage and track changes

[Changelog](/help/changes-management/changelog/) is a crucial feature of Bump.sh, recording the history of all deployments and detailing changes compared to previous versions.

> The [changes comparison](/help/changes-management/changelog/#compare-changelog-entries/) allows you to compare two deployments, regardless of differences in dates or branches, making it easier to update your API users.

Reviewer: ```suggestion
The [changes comparison](/help/changes-management/changelog/#compare-changelog-entries/) allows you to compare two deployed API definitions, regardless of differences in dates or branches.
```

> To offer a smoother experience, you can also integrate [changes management](/help/changes-management/changes-management/) and [tracking integrations](/help/changes-management/integrations/) directly into your existing workflows.

Reviewer: We don't "integrate change management", neither "tracking integration", we should probably rework this sentence.

> And to ensure you don't miss any updates, Bump.sh offers [various notification options](/help/changes-management/changelog/#email--rss-notifications/), whether it's integration into your existing workflows or email notifications, you can be confident that you'll stay informed about important changes.

Reviewer: We talk again about workflow here, it seems redundant with the previous paragraph.
